diff --git a/dist/index.cjs b/dist/index.cjs
index 9c76bd187c78c4aa16510cc798d9932828a801e0..e5ea0d1b423450971fbc041f468cfec1675223b6 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -165,8 +165,9 @@ var DropCursorView = function () {
     key: "dragover",
     value: function dragover(event) {
       if (!this.editorView.editable) return;
+      var rect = this.editorView.dom.getBoundingClientRect();
       var pos = this.editorView.posAtCoords({
-        left: event.clientX,
+        left: rect.left + rect.width / 2,
         top: event.clientY
       });
       var node = pos && pos.inside >= 0 && this.editorView.state.doc.nodeAt(pos.inside);
diff --git a/dist/index.js b/dist/index.js
index 3fc1803920c1ee2565fbb53a10ca502408cb4e4c..655bac385d05e444790063d4666b41694e4ff9ec 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -109,7 +109,8 @@ class DropCursorView {
     dragover(event) {
         if (!this.editorView.editable)
             return;
-        let pos = this.editorView.posAtCoords({ left: event.clientX, top: event.clientY });
+        let rect = this.editorView.dom.getBoundingClientRect()
+        let pos = this.editorView.posAtCoords({ left: rect.left + rect.width / 2, top: event.clientY });
         let node = pos && pos.inside >= 0 && this.editorView.state.doc.nodeAt(pos.inside);
         let disableDropCursor = node && node.type.spec.disableDropCursor;
         let disabled = typeof disableDropCursor == "function" ? disableDropCursor(this.editorView, pos, event) : disableDropCursor;
diff --git a/src/dropcursor.ts b/src/dropcursor.ts
index 9d99133934a7099a54306af1ee115f523fd86881..18c04d79e7c959002f1f6981d6e9ccfe75edff73 100644
--- a/src/dropcursor.ts
+++ b/src/dropcursor.ts
@@ -124,7 +124,8 @@ class DropCursorView {
 
   dragover(event: DragEvent) {
     if (!this.editorView.editable) return
-    let pos = this.editorView.posAtCoords({left: event.clientX, top: event.clientY})
+    let rect = this.editorView.dom.getBoundingClientRect()
+    let pos = this.editorView.posAtCoords({left: rect.left + rect.width / 2, top: event.clientY})
 
     let node = pos && pos.inside >= 0 && this.editorView.state.doc.nodeAt(pos.inside)
     let disableDropCursor = node && node.type.spec.disableDropCursor
